### ISSUE TITLE
[Bug]: Logfire log not wrapped in request span

### ISSUE THREAD

**Issue by ionflow** [BerriAI/litellm#4202]
### What happened?

When making a request using LiteLLM in a fastapi logfire project logs are not wrapped in the span where the request is logged. 
<img width="840" alt="Screenshot 2024-06-14 at 1 12 40 PM" src="https://github.com/BerriAI/litellm/assets/4053903/a1b7236f-a524-4e8c-a189-a5a9abecc17d">

When you collapse the span you can see it is outside of it.
<img width="839" alt="Screenshot 2024-06-14 at 2 42 12 PM" src="https://github.com/BerriAI/litellm/assets/4053903/91db920a-c625-402d-a861-2904ff71c4f2">

I would expect the Chat Completion log to be part of the span like other libraries (instructor/openai client).

**Comment by ishaan-jaff** [comment]
@elisalimli any chance you can take a look at this one / help us with this ?

**Comment by ghost** [comment]
is logfire otel compatible? wondering if we can just migrate to our native otel integration - https://docs.litellm.ai/docs/proxy/logging#logging-proxy-inputoutput-in-opentelemetry-format

> Because Pydantic Logfire is built on [OpenTelemetry](https://opentelemetry.io/), ...

it looks like it's otel compatible, but i can't find docs on logging to it via just otel

**Comment by elisalimli** [comment]
@ishaan-jaff Seems like you are starting a new thread for success_handler. https://github.com/BerriAI/litellm/blob/main/litellm/utils.py#L3867

Logfire groups the spans using the context managers, I am suspicious if the cause of this issue is the threading.

**Comment by elisalimli** [comment]
Replacing starting a new thread with calling the success_handler directly solved the issue. It looks like the started thread doesn't have the context managers.

**Comment by elisalimli** [comment]
> is logfire otel compatible? wondering if we can just migrate to our native otel integration - https://docs.litellm.ai/docs/proxy/logging#logging-proxy-inputoutput-in-opentelemetry-format
> 
> > Because Pydantic Logfire is built on [OpenTelemetry](https://opentelemetry.io/), ...
> 
> it looks like it's otel compatible, but i can't find docs on logging to it via just otel

Would it solve the issue of the non-nested spans?

**Comment by ionflow** [comment]
Does this help: https://docs.pydantic.dev/logfire/guides/onboarding_checklist/integrate/#opentelemetry-instrumentation

**Comment by ishaan-jaff** [comment]
<img width="1280" alt="Screenshot 2024-06-17 at 5 04 26 PM" src="https://github.com/BerriAI/litellm/assets/29436595/ce1b9b4e-6431-4775-9f86-4664395433cc">


Okay I can confirm our OTEL logger works with logfire, looks a lot cleaner too :)

**Comment by ishaan-jaff** [comment]
Hi @elisaliml

This means the logfire integration will start requiring the following dependencies. I want to make sure this does not impact production for you. **what's the best way to roll this out for you ?** 

(We also don't want to add `opentelemetry` as a dependency to litellm pypi ) 

```shell
pip install opentelemetry-api==1.25.0
pip install opentelemetry-sdk==1.25.0
pip install opentelemetry-exporter-otlp==1.25.0
```

**Comment by ishaan-jaff** [comment]
@elisalimli (looks like the mention did not work on the first comment)

**Comment by ishaan-jaff** [comment]
Fix here: https://github.com/BerriAI/litellm/pull/4254

**Comment by ishaan-jaff** [comment]
This should be fixed on 1.40.16 @ionflow 
```
can you try this
litellm.callbacks = ["logfire"]
```

https://docs.litellm.ai/docs/observability/logfire_integration
